include/nbl/video/IGPUDescriptorSet.h

```diff
@@ -99,6 +113,20 @@ class IGPUDescriptorSet : public asset::IDescriptorSet<const IGPUDescriptorSetLa
             return descriptors+localOffset;
         }
 
+        // Same as above, but amortizes lookup if you already have an index
+        inline core::smart_refctd_ptr<asset::IDescriptor>* getDescriptorsIndexed(const asset::IDescriptor::E_TYPE type, const uint32_t bindingStorageIndex) const
```

Comment:
we made the `storage_range_index_t` for a reason, so you couldn't shove a random integer as argument and loose the semantics

note that `binding_number_t` and `storage_range_index_t` both contain a `uint32_t` byt have vastly different meanings.

btw why does the above not use `binding_number_t` and takes `uint32_t` as argument?

Reply:
also why not implement the above in terms of this one

Reply:
btw using typed arguments, would allow you to use the same method name